packages/webarchitect/README.md

```diff
@@ -0,0 +1,251 @@
+```
+
+### Available Options
+
+- `--verbose`: Shows detailed information about each validation, including what files are being checked, what rules are being applied, and the actual content found. Essential for debugging validation failures or understanding exactly what the tool is checking.
+-`--json`: Outputs results in JSON format instead of human-readable text. Required for integrating webarchitect into automated systems, CI/CD pipelines, or other tools that need to programmatically process validation results.
+-`--help`: Displays comprehensive usage information, available options, and examples. Use this command to see the most current information about the tool's capabilities.
+
+### Ruleset Resolution Mechanism
+
+The `<ruleset>` parameter follows a specific resolution order that determines where webarchitect looks for ruleset definitions:
+
+1. **Local Files First** - Webarchitect first checks the current directory for a file matching your ruleset name. If you specify `my-rules`, it looks for `my-rules.json` in the current directory. This allows you to override built-in rulesets or use project-specific validation rules.
+
+2. **Bundled Rulesets Second** - If no local file is found, webarchitect searches its internal collection of built-in rulesets. These include `base`, `base-package`, and `package-react`. Built-in rulesets are distributed with the tool and provide standard validation patterns for common project types.
+
+3. **Remote URLs** - You can also specify a complete URL (starting with `http://` or `https://`) to fetch rulesets from remote locations. Remote rulesets must end with `.json` and be accessible via standard HTTP requests. This enables sharing rulesets across organizations or using centrally managed validation rules.
+
+This resolution mechanism allows teams to start with built-in rulesets, customize them locally as needed, and eventually move to centralized remote rulesets as their validation requirements mature and become standardized across larger organizations.
+
+### Exit Codes
+
+Webarchitect follows standard Unix conventions for exit codes. It returns 0 when all validations pass and 1 when any validation fails. This makes it easy to integrate into scripts and CI/CD pipelines where you want builds to fail if architectural standards aren't met.
+
+## Programmatic API
+
+While the command-line interface covers most use cases, webarchitect also provides a programmatic API for integration into build tools, custom scripts, or other Node.js applications.
+
+### Basic Usage Example
+
+```typescript
+import { validate } from '@canonical/webarchitect';
+import type { ValidationResult } from '@canonical/webarchitect';
+
+// Validate the current directory against a ruleset
+async function checkProjectCompliance(): Promise<void> {
+  try {
+    // The validate function takes a project path and ruleset identifier
+    const results: ValidationResult[] = await validate(process.cwd(), 'base-package');
```

Comment:
Is there any sort of validation metadata that might live at the process level, rather than the rule level? If so, that might change the return of validate() to some object that has multiple properties, like `rules` (the current array of results, for rule-scoped info) and additional properties, for instance `passed` (could save the user from having to manually check all of the rules with `Array.some()`, which is a very common use case)